dcs/task.py

```diff
@@ -1027,6 +1027,47 @@ def __init__(self, from_index=None, to_index=None):
             self.params["nWaypointIndx"] = to_index
 
 
+class WWIIFollowBigFormation(Task):
+    Id = "FollowBigFormation"
+
+    class FormationType(IntEnum):
+        COMBAT_BOX_FOR_OPEN_FORMATION = 18
+        JAVELIN_DOWN = 16
+        COMBAT_BOX = 15
+
+    def __init__(self,
+                 group_id: Optional[int] = None,
+                 position: WWIIFollowBigFormationPosition = WWIIFollowBigFormationPosition(0, 0, 0),
+                 last_wpt_index_flag: bool = False,
+                 last_wpt_index_flag_changed_manually: bool = False,
+                 formation_type: FormationType = FormationType.COMBAT_BOX_FOR_OPEN_FORMATION,
+                 pos_in_box: int = 0,
+                 pos_in_group: int = 0,
+                 pos_in_wing: int = 0,
+                 last_wpt_index: Optional[int] = None) -> None:
+        super().__init__(self.Id)
+
+        self.params = {
+            "pos": position,
```

Comment:
Lua section looks like:

["pos"] =
{
    ["y"] = 0,
    ["x"] = 0,
    ["z"] = 0,
}, -- end of ["pos"]

I thought this would be enough, but let me know if I need to go explicit.

Reply:
I'm pretty sure you do. Have you looked at the serialized miz? Or loaded it in the mission editor? I'm pretty sure you'd see `pos = WWIIBlahBlah(x=a, y=b, z=c)` and not the lua table. I don't see anything that would transform this into a dict. The callers just pass this all up into one giant dict of the whole `Mission`, and that gets passed to `lua.dumps`, which would fall into the `else` branch and just do `str(m.dict())`. `str(position)` is not going to be that lua table you're expecting.

```python
>>> from dataclasses import dataclass
>>> @dataclass
... class Vector3:
...     x: int
...     y: int
...     z: int
...
>>> str(Vector3(0, 0, 0))
'Vector3(x=0, y=0, z=0)'
```

Reply:
@DanAlbert - I understand your points, but mission surprisingly loads just fine with the right data and test actually covers it. What do you want me to do? Keep the naming convention so it belongs to WWIIBigFormation or generic Vector3? I'm assuming Vector3 stays in mapping, while WWIIBigFormationPosition would be moved to WWIIBigFormation class?

Reply:
Have you inspected the miz? (You gave an example above, but I wasn't sure if you meant that was what pydcs produced, or what the ME produced.) I'm pretty surprised it's loading fine, but I may have missed the part that would serialize this correctly (very possible, I'm not that familiar with the serialization code). If you verify in the miz that it's fine then I'm happy without the explicit serialization, but if you haven't inspected it, probably should be explicit, since it might be broken and currently asymptomatic.

As for the name of the class and location, yeah, either name it `Vector3` and keep it here, or keep the name and move it. I don't really care which (in case you just want me to make the call: I'd lean slightly toward renaming `Vector3`, because I expect that'll end up reused sooner or later).

Reply:
@DanAlbert - I did - even I was surprised when I read your post, but that reminded me of one thing. In test, when I added __eq__ to comparing it to pos -- I did it that way because it the task.param["pos"] was a pure dict(), not an object. I understand more about pydcs than I did when I started introducing these patches, so no worries. I'll fix to explicit way so there will be no doubts, but yeah, I did inspect the mission file. I'm actually working on a project that's heavily doing load/modify/save that's why I have to verify everything in .miz too.

Reply:
Ahhhh that makes sense now. Thanks for being thorough 👍